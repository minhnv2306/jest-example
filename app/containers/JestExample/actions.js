/*
 *
 * JestExample actions
 *
 */

import { DEFAULT_ACTION, LOAD_USER_SAGA, LOADED_USER } from './constants';

export function defaultAction() {
	return {
		type: DEFAULT_ACTION,
	};
}

export function loadUser() {
	return {
		type: LOAD_USER_SAGA,
	};
}

export function loadedUser(user) {
	return {
		type: LOADED_USER,
		user,
	};
}
