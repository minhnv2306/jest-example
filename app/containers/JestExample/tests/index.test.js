/**
 *
 * Tests for JestExample
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { JestExample, mapDispatchToProps } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
import configureStore from '../../../configureStore';
import { loadUser } from '../actions';

describe('<JestExample />', () => {
  // it('Expect to not log errors in console', () => {
  //   const spy = jest.spyOn(global.console, 'error');
  //   const dispatch = jest.fn();
  //   render(
  //     <IntlProvider locale={DEFAULT_LOCALE}>
  //       <JestExample dispatch={dispatch} />
  //     </IntlProvider>,
  //   );
  //   expect(spy).not.toHaveBeenCalled();
  // });

  // it('Expect to have additional unit tests specified', () => {
  //   expect(true).toEqual(false);
  // });

  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <JestExample />
        </IntlProvider>
        ,
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the repos on mount if a username exists', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <JestExample onClickButton={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  it('test with user props', () => {
    const submitSpy = jest.fn();
    const user = {
      name: 'MinhNV',
    };
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <JestExample user={user} onClickButton={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onClickButton', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onClickButton).toBeDefined();
      });

      // 100% Funcs
      it('should dispatch loadUser when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onClickButton();
        expect(dispatch).toHaveBeenCalledWith(loadUser());
      });
    });
  });
});
