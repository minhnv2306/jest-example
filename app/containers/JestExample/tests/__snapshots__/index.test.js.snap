// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JestExample /> Should render and match the snapshot 1`] = `
<div>
  <span>
    This is the JestExample container!
  </span>
  <h1>
     Hello I am Minh 
  </h1>
  <button
    type="button"
  >
    Click me
  </button>
</div>
`;
